build/lua/test/socket.lua

```diff
@@ -0,0 +1,54 @@
+local function check_index(socket)
```

Comment:
Could you move all .lua files that are within the _build_ directory but not related to the build process to the top level _lua_ directory or the top level _tests_ directory ?